realm/src/main/java/io/realm/Realm.java

```diff
@@ -895,6 +904,80 @@ public void writeCopyTo(File destination) throws IOException {
         return result;
     }
 
+    /**
+     * Converts a RealmObject that has been created using it's standard constructors instead of
+     * {@link #createObject(Class)} to a RealmObject that is managed by the Realm.
+     *
+     * To persist any further changes, the returned object must be be used.
+     *
+     * @param object RealmObject to copy to the Realm.
+     * @return The converted RealmObject that has all its properties managed by the Realm.
+     *
+     * @throws io.realm.exceptions.RealmException if object has already been added to the Realm.
+     */
+    public <E extends RealmObject> E copyToRealm(E object) {
+        if (object == null) {
+            throw new IllegalArgumentException("Null objects cannot be copied into Realm.");
+        }
+
+        // Object is already in this Realm
+        if (object.realm == this) {
```

Comment:
We compute an id from the path. we will have to compute one for in-memory realms as well so that might be a good comparison item